### ISSUE TITLE
Never use cache, always reindex

### ISSUE THREAD

**Issue by deafwolf** [MaskRay/ccls#682]
### Observed behavior

run ccls --index=.
will parse/store almost all source files (timestamp will be updated).
all source files are not changed.

when I open emacs, then open any source file, it output many logs in *ccls::stderr* and stop after 30 minutes.
it's same when I repoen emacs.

the log in *ccls::stderr* like this, why?

14:12:18 indexer4         pipeline.cc:378 I store index for /home/deafwolf/thirdparty/scylla/auth/common.hh (delta: 0)
14:12:25 indexer0         pipeline.cc:378 I store index for /home/deafwolf/thirdparty/scylla/auth/common.hh (delta: 1)
14:22:57 indexer1         pipeline.cc:340 I parse /home/deafwolf/thirdparty/scylla/auth/common.hh

14:50:27 indexer3         pipeline.cc:378 I store index for /usr/include/boost/mpl/bitxor.hpp (delta: 1)
14:50:27 indexer3         pipeline.cc:378 I store index for /usr/include/boost/uuid/entropy_error.hpp (delta: 1)
14:50:27 indexer3         pipeline.cc:378 I store index for /usr/include/boost/uuid/detail/random_provider_detect_platform.hpp (delta: 1)
14:50:27 indexer3         pipeline.cc:378 I store index for /usr/include/boost/random/detail/generator_seed_seq.hpp (delta: 1)
14:50:27 indexer3         pipeline.cc:378 I store index for /usr/include/boost/random/detail/polynomial.hpp (delta: 1)
14:50:27 indexer3         pipeline.cc:378 I store index for /usr/include/boost/random/detail/uniform_int_float.hpp (delta: 1)

### Expected behavior

the source file which has been indexed will not be reindexed.

### Steps to reproduce

git clone git@github.com:scylladb/scylla.git
cd scylla
git submodule update --init --recursive
./configure.py --mode dev --enable-dpdk --with scylla # maybe need install some depends
ninja -t compdb > compile_commands.json
ccls --index=.
ccls --index=.

open source file with emacs will also see the same log.

### System information

* ccls version: 4cf16a7e5a196cdd062054a8aaf896ac7e955bd7 and 0.20190823.6
* clang version: clang version 10.0.0 (Fedora 10.0.0-2.fc32)
* OS: fedora32
* Editor: emacs-26.3-2.fc32.src.rpm
* Client: emacs ccls 20200710.1810

**Comment by deafwolf** [comment]
I config the emacs to avoid the time cost.
(setq ccls-initialization-options '(:index (:initialBlacklist (".")))) # from https://github.com/MaskRay/ccls/issues/180

Maybe this is also a problem, please confirm it or not.

**Comment by bastianbeischer** [comment]
Confirmed, I think reverting cb08df44aa552d3120e2e1d317f9aaafdf2cfd65 fixes it.

**Comment by deafwolf** [comment]
> Confirmed, I think reverting [cb08df4](https://github.com/MaskRay/ccls/commit/cb08df44aa552d3120e2e1d317f9aaafdf2cfd65) fixes it.

Maybe not，the tag 0.20190823.6 also has this problem.

**Comment by bastianbeischer** [comment]
> > Confirmed, I think reverting [cb08df4](https://github.com/MaskRay/ccls/commit/cb08df44aa552d3120e2e1d317f9aaafdf2cfd65) fixes it.
> 
> Maybe not，the tag 0.20190823.6 also has this problem.

IIRC a problem with the same symptoms existed last year, I would not need to check but that might be the reason why the tag you mentioned also has this problem. Did you try reverting the commit I mentioned?

**Comment by deafwolf** [comment]
commit ee2ff85fcd42d895006b4e792d391982bbd48b92 (HEAD -> revert)
Author: deafwolf <deafwolfdruid@gmail.com>
Date:   Tue Aug 4 14:50:44 2020 +0800

    Revert "Support workspaces as symlinks and symlinks within a workspace"
    
    This reverts commit cb08df44aa552d3120e2e1d317f9aaafdf2cfd65.

commit 4cf16a7e5a196cdd062054a8aaf896ac7e955bd7 (origin/master, origin/HEAD, master)
Author: Fangrui Song <i@maskray.me>
Date:   Sat Jul 25 08:30:18 2020 -0700

    Adapt llvmorg-12-init-946-g10b1b4a231a: add find_package(ZLIB)

commit cb08df44aa552d3120e2e1d317f9aaafdf2cfd65
Author: Fangrui Song <i@maskray.me>
Date:   Sun Jul 19 17:03:16 2020 -0700

    Support workspaces as symlinks and symlinks within a workspace
    
    Fix #639

I think there is no effect, all files in .ccls-cache are changed.

14:54:57 indexer2         pipeline.cc:397 I store index for /home/deafwolf/thirdparty/scylla/auth/common.hh (delta: 0)
14:55:03 indexer0         pipeline.cc:397 I store index for /home/deafwolf/thirdparty/scylla/auth/common.hh (delta: 0)
15:10:59 indexer3         pipeline.cc:386 I parse /home/deafwolf/thirdparty/scylla/auth/common.hh error:2 redefinition of 'once_among_shards'

**Comment by bastianbeischer** [comment]
Hm, I tested the latest master again without any reverted commits and I can not reproduce the problem anymore with my code. Weird, because I'm sure I had this problem yesterday... Anyway, reverting the commit I mentioned above doesn't seem to make a difference after all - sorry for the noise.

**Comment by deafwolf** [comment]
Never mind. I think your advice is useful, the ccls with reverting cost less time.
I'm not  familiar with ccls, just a user. Scylladb is too large, I'd like to provide a minimal code to reproduce the problem, but I don't know how to. any advice?

**Comment by findNextStep** [comment]
i gauss you use ccls from homebrew. it`s too old to use. build from source code with master should work

**Comment by bastianbeischer** [comment]
> ```
> 14:54:57 indexer2         pipeline.cc:397 I store index for /home/deafwolf/thirdparty/scylla/auth/common.hh (delta: 0)
> 14:55:03 indexer0         pipeline.cc:397 I store index for /home/deafwolf/thirdparty/scylla/auth/common.hh (delta: 0)
> 14:55:12 indexer0         pipeline.cc:397 I store index for /home/deafwolf/thirdparty/scylla/seastar/include/seastar/http/common.hh (delta: 1)
> 15:10:59 indexer3         pipeline.cc:386 I parse /home/deafwolf/thirdparty/scylla/auth/common.hh error:2 redefinition of 'once_among_shards'
> ```

Is it possible that your code simply doesn't compile? Or maybe the preprocessing is different between ccls and your actual compilation?

**Comment by deafwolf** [comment]
* about the reindex
I found that if generate compile_commands.json with rule: `ninja -t compdb cxx.dev`, ccls will not reindex. Looks this issue has been fixed, though I don't exactly know why, `ninja -t compdb` will generate some command such as code gen, maybe this is the reason.

* about index error
scylladb can only be compiled by gcc, I think there is some compatible problems.

* about ccls version
I tried two version, release 0.20190823.6, and compile from master  4cf16a7.
Both of them has the same problem. The different is that release version will not log any error but master version will do.
If a error occurs, is the file will be completely indexed?
